apps/web/src/hooks/useSafeTokenAllocation.ts

```diff
@@ -116,41 +108,41 @@ const useSafeTokenAllocation = (): AsyncResult<Vesting[]> => {
   return useAsync<Vesting[] | undefined>(async () => {
     if (!safeAddress) return
     return Promise.all(
-      await fetchAllocation(chainId, safeAddress).then((allocations) =>
-        allocations.map((allocation) => completeAllocation(allocation)),
-      ),
+      await fetchAllocation(chainId, safeAddress).then((allocations) => completeAllocations(allocations)),
     )
     // If the history tag changes we could have claimed / redeemed tokens
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [chainId, safeAddress, safe.txHistoryTag])
 }
 
-const fetchTokenBalance = async (chainId: string, safeAddress: string): Promise<string> => {
+const fetchTokenBalances = async (chainId: string, safeAddress: string): Promise<[bigint, bigint]> => {
   try {
     const web3ReadOnly = getWeb3ReadOnly()
     const safeTokenAddress = getSafeTokenAddress(chainId)
-    if (!safeTokenAddress || !web3ReadOnly) return '0'
-
-    return await web3ReadOnly.call({
-      to: safeTokenAddress,
-      data: tokenInterface.encodeFunctionData('balanceOf', [safeAddress]),
-    })
-  } catch (err) {
-    throw Error(`Error fetching Safe Token balance:  ${err}`)
-  }
-}
-const fetchLockingContractBalance = async (chainId: string, safeAddress: string): Promise<string> => {
-  try {
-    const web3ReadOnly = getWeb3ReadOnly()
     const safeLockingAddress = getSafeLockingAddress(chainId)
-    if (!safeLockingAddress || !web3ReadOnly) return '0'
 
-    return await web3ReadOnly.call({
-      to: safeLockingAddress,
-      data: safeLockingInterface.encodeFunctionData('getUserTokenBalance', [safeAddress]),
-    })
+    if (!web3ReadOnly || !safeTokenAddress || !safeLockingAddress) return [BigInt(0), BigInt(0)]
+
+    const calls = [
+      {
+        to: safeTokenAddress,
+        data: tokenInterface.encodeFunctionData('balanceOf', [safeAddress]),
+      },
+      {
+        to: safeLockingAddress,
+        data: safeLockingInterface.encodeFunctionData('getUserTokenBalance', [safeAddress]),
+      },
+    ]
+
+    const results = await multicall(web3ReadOnly, calls)
+
+    if (!results[0].success || !results[1].success) {
+      throw new Error('Failed to fetch token balances')
+    }
+
+    return [BigInt(results[0].returnData), BigInt(results[1].returnData)]
```

Comment:
nit: For better readability I would suggest to destructure results into named consts like you do in `useSafeVotingPower`

```
const [tokenBalance, lockingContractBalance] = await multicall(...)

if (!tokenBalance.success || !lockingContractBalance.success) ...
```